src/orders/components/OrderFulfillment/TableLine.tsx

```diff
@@ -0,0 +1,124 @@
+import { makeStyles, TableCell, TableRow } from "@material-ui/core";
+import Money from "@saleor/components/Money";
+import Skeleton from "@saleor/components/Skeleton";
+import TableCellAvatar, {
+  AVATAR_MARGIN
+} from "@saleor/components/TableCellAvatar";
+import { maybe } from "@saleor/misc";
+import {
+  OrderDetails_order_fulfillments_lines,
+  OrderDetails_order_lines
+} from "@saleor/orders/types/OrderDetails";
+import React from "react";
+
+const useStyles = makeStyles(
+  theme => ({
+    clickableRow: {
+      cursor: "pointer"
+    },
+    colName: {
+      width: "auto"
+    },
+    colNameLabel: {
+      marginLeft: AVATAR_MARGIN
+    },
+    colPrice: {
+      textAlign: "right",
+      width: 120
+    },
+    colQuantity: {
+      textAlign: "center",
+      width: 120
+    },
+    colSku: {
+      textAlign: "right",
+      textOverflow: "ellipsis",
+      width: 120
+    },
+    colTotal: {
+      textAlign: "right",
+      width: 120
+    },
+    infoLabel: {
+      display: "inline-block"
+    },
+    infoLabelWithMargin: {
+      marginBottom: theme.spacing()
+    },
+    infoRow: {
+      padding: theme.spacing(2, 3)
+    },
+    orderNumber: {
+      display: "inline",
+      marginLeft: theme.spacing(1)
+    },
+    statusBar: {
+      paddingTop: 0
+    },
+    table: {
+      tableLayout: "fixed"
+    }
+  }),
+  { name: "TableLine" }
+);
+
+interface TableLineProps {
+  line: OrderDetails_order_fulfillments_lines | OrderDetails_order_lines;
+  isOrderLine?: boolean;
+}
+
+const TableLine: React.FC<TableLineProps> = ({
+  line: lineData,
+  isOrderLine = false
+}) => {
+  const classes = useStyles({});
+
+  if (!lineData) {
+    return <Skeleton />;
+  }
+
+  const line = isOrderLine
+    ? ({
+        ...lineData,
+        orderLine: lineData
+      } as OrderDetails_order_fulfillments_lines)
+    : (lineData as OrderDetails_order_fulfillments_lines);
+
+  return (
+    <TableRow className={classes.clickableRow} hover key={line.id}>
+      <TableCellAvatar
+        className={classes.colName}
+        thumbnail={maybe(() => line.orderLine.thumbnail.url)}
+      >
+        {maybe(() => line.orderLine.productName) || <Skeleton />}
+      </TableCellAvatar>
+      <TableCell className={classes.colSku}>
+        {line?.orderLine.productSku || <Skeleton />}
+      </TableCell>
+      <TableCell className={classes.colQuantity}>
+        {line?.quantity || <Skeleton />}
```

Comment:
quantity is string or number? Wonder if skeleton will be rendered if quantity is 0

Reply:
Tbh noticed this (and some other stuff like that in this component) but didn't check, because this is just copied pasted from the main component. I wanted to wait with more refactoring of this because seems there's a situation that the designs don't cover - when a fulfillment line in a fulfillment has no orderline, because the purchased item has been removed from the shop DB altogether. So waiting for the designs from @pwgryglak and then I think it'll make more sense to refactor this component.